airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py

```diff
@@ -52,48 +58,52 @@ def send_request(self, query: str) -> SearchGoogleAdsResponse:
     @staticmethod
     def get_fields_from_schema(schema: Mapping[str, Any]) -> List[str]:
         properties = schema.get("properties")
-
         return [*properties]
 
     @staticmethod
-    def convert_schema_into_query(schema: Mapping[str, Any], report_name: str, from_date: str, to_date: str) -> str:
+    def convert_schema_into_query(
+        schema: Mapping[str, Any], report_name: str, from_date: str = None, to_date: str = None, cursor_field: str = None
+    ) -> str:
         from_category = REPORT_MAPPING[report_name]
         fields = GoogleAds.get_fields_from_schema(schema)
         fields = ",\n".join(fields)
 
-        query = Template(
-            """
-          SELECT
-            $fields
-          FROM $from_category
-          WHERE segments.date > '$from_date'
-            AND segments.date < '$to_date'
-          ORDER BY segments.date
-      """
-        )
-        query = query.substitute(fields=fields, from_category=from_category, from_date=from_date, to_date=to_date)
+        query_template = "SELECT {fields} FROM {from_category} "
+        substitute_params = dict(fields=fields, from_category=from_category)
+
+        if cursor_field:
+            query_template += "WHERE {cursor_field} > '{from_date}' AND {cursor_field} < '{to_date}' ORDER BY {cursor_field}"
+            substitute_params.update(dict(from_date=from_date, to_date=to_date, cursor_field=cursor_field))
 
+        query = query_template.format(**substitute_params)
         return query
```

Comment:
```suggestion
        return query_template.format(**substitute_params)
```

Why create `query` variable if we can just return the result?